src/components/Button/Button.style.ts

```diff
@@ -59,14 +65,23 @@ export const buttonStyle = ({
 
 export const buttonSpanStyle = ({
   icon,
+  iconAlign,
   size,
-  hasChildren,
 }: RequiredProperties<Props & { hasChildren: boolean }>) => (theme: Theme) => ({
   display: icon ? 'flex' : 'block',
-  flexDirection: (icon ? 'row' : 'column') as FlexDirectionProperty,
+  flexDirection: (iconAlign === 'right' ? 'row-reverse' : 'row') as FlexDirectionProperty,
   alignItems: icon ? 'center' : 'flex-start',
-  '> :first-child': {
-    marginLeft: icon ? (size === 'sm' ? theme.spacing.sm : theme.spacing.md) : 0,
-    marginRight: hasChildren ? theme.spacing.sm : 0,
-  },
+  marginLeft: iconAlign === 'right' ? (size === 'sm' ? theme.spacing.sm : theme.spacing.md) : 0,
 });
+
+export const iconStyle = ({
+  size,
+  iconAlign,
+}: RequiredProperties<Props & { hasChildren: boolean }>) => (theme: Theme) => {
+  const margin = size === 'sm' ? theme.spacing.sm : theme.spacing.md;
+
+  return {
+    marginLeft: iconAlign === 'left' ? margin : theme.spacing.sm,
```

Comment:
You could refactor this line of code into an arrow function in the iconStyle's scope since the line below is almost identical.

example:
```
const getMargin = (alignPosition) => iconAlign === alignPosition ? margin : theme.spacing.sm;
....
return {
marginLeft: getMargin('left'),
marginRight: getMargin('right')
}
```